packages/babel-plugin-transform-block-scoping/src/index.js

```diff
@@ -519,20 +517,18 @@ class BlockScoping {
 
     // remap loop heads with colliding variables
     if (this.loop) {
-      for (const name of Object.keys(outsideRefs)) {
-        const id = outsideRefs[name];
+      for (const name of outsideRefs.keys()) {
+        const id = outsideRefs.get(name);
 
         if (
           this.scope.hasGlobal(id.name) ||
           this.scope.parentHasBinding(id.name)
         ) {
-          delete outsideRefs[id.name];
-          delete this.letReferences[id.name];
+          outsideRefs.delete(id.name);
 
           this.scope.rename(id.name);
 
-          this.letReferences[id.name] = id;
-          outsideRefs[id.name] = id;
+          outsideRefs.set(id.name, id);
```

Comment:
Aha.  I should have noticed that these variables (`this.letReferences`, `outsideRefs`) aren't referring to the same object (I did think it was strange, so I de-duplicated them; but now I realize they're not duplicates in the first place..)